superset/views/database/api.py

```diff
@@ -274,6 +351,139 @@ def table_metadata(self, database: Database, table_name: str, schema_name: str):
         self.incr_stats("success", self.table_metadata.__name__)
         return self.response(200, **table_info)
 
+    @expose("/<int:pk>/tables/<string:schema_name>/<string:substr>/", methods=["GET"])
+    @expose(
+        "/<int:pk>/tables/<string:schema_name>/<string:substr>/<force_refresh>/",
+        methods=["GET"],
+    )
+    @protect()
+    @safe
+    @event_logger.log_this
+    def tables(  # pylint: disable=invalid-name,too-many-locals
```

Comment:
I'm starting to wonder if we want to do this migration now as part of the security fix PR? This looks like we've pulled the method definition over almost as-is, but as `pylint` is showing us, this really could use a refactor. Would it be better to have one PR that adds the security fix, then a second that refactors the method into the new API?

Another approach here could be to "strangle" the current implementation by proxying to the current implementation, then refactoring it over time.

Reply:
Agreed I'll break this into 2 PR's